### PR TITLE
Refactor reactive ready approach

### PR DESCRIPTION
**Pull request by rijk**
- Fixed an infinite loop when calling `handle.ready()`
- Renamed the subscription handle ref to `subscription` to avoid confusion
- Had to move the handle out of the ref, otherwise Typescript got confused by the circular reference (`stop` and `ready` use `ref.subscription`) and you lose typings for `ref`
- Similarly, I had to change the `?.stop()` and `?.ready()` to full checks to silence Typescript warnings

### CODE REVIEW

File: packages/react-mongo/react-mongo.ts [hunk 1]
```diff
@@ -32,9 +25,9 @@ const useSubscriptionClient = (factory: () => Meteor.SubscriptionHandle | void,
     // it stops the inner one.
     const computation = Tracker.nonreactive(() => (
       Tracker.autorun(() => {
-        refs.handle = factory()
-        if (!refs.handle) return
-        if (refs.updateOnReady && refs.handle.ready()) {
+        refs.subscription = factory()
+        if (!refs.subscription) return
+        if (refs.updateOnReady && refs.subscription.ready()) {
```
Comment (yched) [inline packages/react-mongo/react-mongo.ts:30]: does that really work ?
updateOnReady starts as false, so the tracker computation doesn't run subscription.ready() and thus doesn't register it as a reactive dep.
Then if the user calls the ready() func we give him, updateOnReady is changed to true, but that doesn't make the tracker computation rerun and become reactive to the ready state change ?

Comment (rijk) [reply packages/react-mongo/react-mongo.ts:30]: Ah, good point. I actually swapped these around (`if (refs.subscription.ready() && refs.updateOnReady)`) to fix this, but I took it out of this PR as to not change too much. Do you think that would work if it is swapped?

Comment (yched) [reply packages/react-mongo/react-mongo.ts:30]: yep, sounds like that should solve it.
the "become reactive once you actually read it" pattern is not super intuitive IMO :-D 
as discussed in the main PR, I tend to think leaving ready(reactivity out of the hook might be simpler...

Comment (rijk) [reply packages/react-mongo/react-mongo.ts:30]: Agreed. I'm working on a PR now, will let you know when it's ready so you can review.

**Pull request closed without merging**